rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -225,10 +225,20 @@ where
                 .await?
                 .saturating_add(U256::from(GAS_ESTIMATE_BUFFER))
         };
+        let chain_id = match self.provider.get_chainid().await {
```

Comment:
Is this needed? Seems like it is not polygon specific but handling something that should have already been?

Reply:
1933 is pretty explicit about it that it is polygon specific no?

Reply:
No, I mean, if you move this match statement to after the `estimate_eip_1559_fees` call you could just `.expect("all 1559 chains should have a chain id")` rather than using a whole match statement.